recipes/soplex/all/conanfile.py

```diff
@@ -0,0 +1,124 @@
+from conan import ConanFile
+from conan.tools.apple import fix_apple_shared_install_name
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.files import collect_libs, copy, get, replace_in_file
+from conan.tools.microsoft import check_min_vs, is_msvc
+from conan.tools.scm import Version
+from os.path import join
+
+required_conan_version = ">=1.53.0"
+
+
+class SoPlexConan(ConanFile):
+    name = "soplex"
+    description = "SoPlex linear programming solver"
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://soplex.zib.de"
+    topics = ("simplex", "solver", "linear", "programming")
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "BOOST": [True, False],
+        "GMP": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "BOOST": False,
+        "GMP": False
+    }
+
+    @property
+    def _min_cppstd(self):
+        return 14
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "5",
+            "clang": "4",
+            "apple-clang": "7",
+        }
+
+    def _determine_lib_name(self):
+        if self.options.shared:
+            return "soplexshared"
+        elif self.options.get_safe("fPIC"):
+            return "soplex-pic"
+        else:
+            return "soplex"
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        check_min_vs(self, 191)
+        if not is_msvc(self):
+            minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+            if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+                raise ConanInvalidConfiguration(
+                    f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+                )
+        if is_msvc(self) and self.options.shared:
+            raise ConanInvalidConfiguration(f"{self.ref} can not be built as shared on Visual Studio and msvc.")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+        replace_in_file(self, "CMakeLists.txt", "GMP_INCLUDE_DIRS", "gmp_INCLUDE_DIRS")
+        replace_in_file(self, "CMakeLists.txt", "GMP_LIBRARIES", "gmp_LIBRARIES")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def requirements(self):
+        self.requires("zlib/1.2.13", transitive_headers=True)
+        if self.options.GMP:
```

Comment:
Our normal practice would be, the options are a dict no an object with properties so this is a touch more clear :) pure style 

```suggestion
        if self.options["with_gmp"]:
```

Reply:
I think that is the syntax to access the options of requirements, not to access the options my own package

Reply:
You are spot on, the casing with the orange syntax was throwing me off :)